src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -342,7 +342,7 @@ object LDPrune {
 
     val contigStartPartitions =
       partitionIndices.filter { i =>
-        i == partitionIndices(0) || i == partitionIndices.last || rangeBounds(i).contig != rangeBounds(i - 1).contig
+        i == partitionIndices(0) || i == partitionIndices.last || rangeBounds(i).start.asInstanceOf[UnsafeRow].getAs[Locus](0).contig != rangeBounds(i).end.asInstanceOf[UnsafeRow].getAs[Locus](0).contig
```

Comment:
I think the first/last checks are no longer necessary here either.  @jigold can you confirm?

Reply:
I think the first check is still needed. What happens if you have the first partition is only chromosome 1? Then the contig start for chr 1 is still in partition 0, but the range bounds don't have different chromosomes.

Also, what if the start range bound is the start of a new chromosome? I think we should be comparing the start of one partition to the end of the previous partition as well.

This code is horrible! So glad Meredith will eventually get rid of it.

Reply:
Is there any circumstance where the partition indices on an OrderedRVD aren't just 0... n-1? I feel like that might cut down on some of the logic, if so.

Reply:
No.  I was going to suggest ripping that out.  Throw in an assert if you want to be extra safe.